### DISCUSSION TITLE
Is it ok to alter `__fields_set__` value of an instance?

### DISCUSSION THREAD

**Discussion by sarangsbabu367** [pydantic/pydantic#3797] [Question]
```python
from typing import Optional
from pydantic import BaseModel, StrictInt, StrictStr

class User(BaseModel):
    id: Optional[StrictInt] = None
    name: Optional[StrictStr] = None
    title: Optional[StrictStr] = None


def first(obj: User):
    # some logic
    obj.__fields_set__.discard("title")
    second(obj)
    obj.__fields_set__.add("title")


def second(obj: User):
    obj_dict = obj.dict(exclude_unset=True)
    if "title" not in obj_dict:
        # some logic
        pass

first(User(id=123, name="Emma", title="Scientist"))
```

Here, i have 2 functions(`first`, `second`). `first` function will get a `User` object and after some logic, that object is passed to `second` function. But the `first` function doesnt intend to do the logic based on `title` property in `second` function. So `title` property is discarded before calling `second` and added back later.

Is there any problem with this approach? Is it required to create a new instance without `title` property in function `first` before calling function `second`? Is there any other way to make a *field* unset after creating an instance?

**Comment by vadim-su** [comment]
Thanks for the idea. I have a set of classes with a discriminator field. And I have to send this field to a web service, but I don't need to set this field manually. Your solution looks good.
